### ISSUE TITLE
vm2 depreceated with security warnings

### ISSUE THREAD

**Issue by mishabruml** [DataDog/serverless-plugin-datadog#409]
https://www.npmjs.com/package/vm2 

> The library contains critical security issues and should not be used for production! The maintenance of the project has been discontinued. Consider migrating your code to isolated-vm.

detected via snyk

```
✗ Remote Code Execution (RCE) [Critical Severity][https://security.snyk.io/vuln/SNYK-JS-VM2-5772823] in vm2@3.9.19
    introduced by serverless-plugin-datadog@5.36.0 > @datadog/datadog-ci@2.16.1 > proxy-agent@5.0.0 > pac-proxy-agent@5.0.0 > pac-resolver@5.0.1 > degenerator@3.0.4 > vm2@3.9.19
  No upgrade or patch available
  ✗ Remote Code Execution (RCE) [Critical Severity][https://security.snyk.io/vuln/SNYK-JS-VM2-5772825] in vm2@3.9.19
    introduced by serverless-plugin-datadog@5.36.0 > @datadog/datadog-ci@2.16.1 > proxy-agent@5.0.0 > pac-proxy-agent@5.0.0 > pac-resolver@5.0.1 > degenerator@3.0.4 > vm2@3.9.19
```

https://security.snyk.io/vuln/SNYK-JS-VM2-5772823
https://security.snyk.io/vuln/SNYK-JS-VM2-5772825

**Comment by psinglet** [comment]
Issues with no direct upgrade or patch:
  ✗ Remote Code Execution (RCE) [Critical Severity][https://security.snyk.io/vuln/SNYK-JS-VM2-5772823] in vm2@3.9.19
    introduced by serverless-plugin-datadog@5.37.0 > @datadog/datadog-ci@2.17.1 > proxy-agent@6.2.2 > pac-proxy-agent@6.0.4 > pac-resolver@6.0.2 > degenerator@4.0.4 > vm2@3.9.19
  No upgrade or patch available
  ✗ Remote Code Execution (RCE) [Critical Severity][https://security.snyk.io/vuln/SNYK-JS-VM2-5772825] in vm2@3.9.19
    introduced by serverless-plugin-datadog@5.37.0 > @datadog/datadog-ci@2.17.1 > proxy-agent@6.2.2 > pac-proxy-agent@6.0.4 > pac-resolver@6.0.2 > degenerator@4.0.4 > vm2@3.9.19
  No upgrade or patch available
  ✗ Regular Expression Denial of Service (ReDoS) [High Severity][https://security.snyk.io/vuln/SNYK-JS-WORDWRAP-3149973] in word-wrap@1.2.3
    introduced by serverless-plugin-datadog@5.37.0 > @datadog/datadog-ci@2.17.1 > proxy-agent@6.2.2 > pac-proxy-agent@6.0.4 > pac-resolver@6.0.2 > degenerator@4.0.4 > escodegen@1.14.3 > optionator@0.8.3 > word-wrap@1.2.3
  No upgrade or patch available

**Comment by DarcyRaynerDD** [comment]
Hi @mishabruml, thanks for the report. We are looking at fixing the issue downstream in datadog-ci, which is where we are picking up this transitive dependency from.

**Comment by mishabruml** [comment]
> Hi @mishabruml, thanks for the report. We are looking at fixing the issue upstream in datadog-ci, which is where we are picking up this transitive dependency from.

Good news! I'm impressed that datadog as an org seem to pick these issues up pretty quickly, thank you, keep it up 👍

**Comment by DarcyRaynerDD** [comment]
This is now fixed in [v5.38.1 ](https://github.com/DataDog/serverless-plugin-datadog/releases/tag/v5.38.1)

**Comment by mishabruml** [comment]
🚤
